src/components/Layout/PhoneNumberInput.tsx

```diff
@@ -63,6 +66,31 @@ export const PhoneNumberInput: FunctionComponent<{
   // eslint-disable-next-line @typescript-eslint/no-empty-function
   onSubmit = () => {}
 }) => {
+  const formatPhoneNumber = (
+    mobileNumberValue: string,
+    region = "SG"
+  ): string => {
+    try {
+      return phoneUtil.format(
+        phoneUtil.parse(mobileNumberValue, region),
+        PhoneNumberFormat.NATIONAL
+      );
+    } catch (e) {
+      if (value && value?.length < mobileNumberValue.length) {
+        return value;
+      }
+      return mobileNumberValue;
+    }
+  };
+  const [value, setValue] = useState<string>(
+    formatPhoneNumber(mobileNumberValue.replace(" ", ""))
+  );
+  useEffect(() => {
+    if (value) {
+      onChangeMobileNumber(value.replace(" ", ""));
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [value]);
```

Comment:
I didn't think of exhaustive deps, let's change this back to how you did it before:

```suggestion
  const onHandleChangePhoneNumber = (mobileNumberValue: string) => {
    const formattedPhoneNumber = formatPhoneNumber(mobileNumberValue);
    onChangeMobileNumber(formattedPhoneNumber);
    setMobileNumber(formattedPhoneNumber);
  }
```

Reply:
👍